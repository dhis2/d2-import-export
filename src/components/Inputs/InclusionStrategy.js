import React from 'react'
import i18n from '@dhis2/d2-i18n'
import { RadioGroupField } from '../index'

const inclusionStrategyOptions = [
    {
        value: 'NON_NULL',
        label: i18n.t('NON NULL: Includes properties which are not null', {
            nsSeparator: '>',
        }),
    },
    {
        value: 'ALWAYS',
        label: i18n.t('ALWAYS: Include all properties', {
            nsSeparator: '>',
        }),
    },
    {
        value: 'NON_EMPTY',
        label: i18n.t('NON EMPTY: Include non-empty properties', {
            nsSeparator: '>',
        }),
    },
]
const defaultInclusionStrategyOption = inclusionStrategyOptions[0].value

const NAME = 'inclusionStrategy'
const DATATEST = 'input-inclusion-strategy'
const LABEL = i18n.t('Inclusion strategy')
const HELPTEXT = i18n.t('Controls which properties to include')

const InclusionStrategy = () => (
    <RadioGroupField
        name={NAME}
        label={LABEL}
        options={inclusionStrategyOptions}
        helpText={HELPTEXT}
        dataTest={DATATEST}
        vertical
    />
)

export { InclusionStrategy, defaultInclusionStrategyOption }
