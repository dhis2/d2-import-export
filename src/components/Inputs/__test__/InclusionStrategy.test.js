import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { InclusionStrategy } from '../InclusionStrategy'

jest.mock('react-final-form')

describe('Input component - InclusionStrategy', () => {
    it('should render correctly', () => {
        const file = shallow(<InclusionStrategy />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
