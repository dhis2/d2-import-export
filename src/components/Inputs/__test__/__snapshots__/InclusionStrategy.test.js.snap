// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - InclusionStrategy should render correctly 1`] = `
<RadioGroup
  dataTest="input-inclusion-strategy"
  label="Inclusion strategy"
  name="inclusionStrategy"
  options={
    Array [
      Object {
        "label": "Non Null",
        "value": "NON_NULL",
      },
      Object {
        "label": "Always",
        "value": "ALWAYS",
      },
      Object {
        "label": "Non Empty",
        "value": "NON_EMPTY",
      },
    ]
  }
/>
`;
